### PR TITLE
added Error and Errorf fix #15

### LINKED ISSUE #15: Logger API `Error(...)` method
A convenience function to help with error handling could be useful.

```go
if err != nil {
    log.From(ctx).Error("Couldn't foo", err)
    return err
}
```
Should produce something like

```sh
msg="Couldn't foo" error_message="connection timed out"
```

### PR DESCRIPTION
**Pull request by nofun97**

### CODE REVIEW

**Comment by nofun97** [comment]
rebased and resolved the conflicts

**Comment by nofun97** [comment]
accidentally added IDE settings

File: updated_readme.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# Design Choices
```
Comment (anzdaddy) [inline updated_readme.md]: What's this file for? The name is a little odd.

Comment (nofun97) [reply updated_readme.md]: My bad, accidentally added unnecessary things again. Will delete soon.

**Pull request merged**